cmd/agent/app/server.go

```diff
@@ -99,13 +101,19 @@ func (a *Agent) runHealthServer(o *options.GrpcProxyAgentOptions) error {
 		MaxHeaderBytes: 1 << 20,
 	}
 
-	go func() {
-		err := healthServer.ListenAndServe()
-		if err != nil {
-			klog.ErrorS(err, "health server could not listen")
-		}
-		klog.V(0).Infoln("Health server stopped listening")
-	}()
+	labels := runpprof.Labels(
+		"core", "healthListener",
+		"port", strconv.Itoa(o.HealthServerPort),
+	)
+	go runpprof.Do(context.Background(), labels, func(context.Context) {
+		func() {
```

Comment:
> if there's a reason not to include the labels here

A little confused by what you mean. The labels are included here. Are you suggesting they shouldn't?
-----------+-------------------------------------------------------
      core:  healthListener
      port:  8093
         1   runtime.gopark /home/wfender/.gvm/gos/go1.17.12/src/runtime/proc.go:366
             runtime.netpollblock /home/wfender/.gvm/gos/go1.17.12/src/runtime/netpoll.go:523
             internal/poll.runtime_pollWait /home/wfender/.gvm/gos/go1.17.12/src/runtime/netpoll.go:303
             internal/poll.(*pollDesc).wait /home/wfender/.gvm/gos/go1.17.12/src/internal/poll/fd_poll_runtime.go:84
             internal/poll.(*pollDesc).waitRead /home/wfender/.gvm/gos/go1.17.12/src/internal/poll/fd_poll_runtime.go:89 (inline)
             internal/poll.(*FD).Accept /home/wfender/.gvm/gos/go1.17.12/src/internal/poll/fd_unix.go:402
             net.(*netFD).accept /home/wfender/.gvm/gos/go1.17.12/src/net/fd_unix.go:173
             net.(*TCPListener).accept /home/wfender/.gvm/gos/go1.17.12/src/net/tcpsock_posix.go:140
             net.(*TCPListener).Accept /home/wfender/.gvm/gos/go1.17.12/src/net/tcpsock.go:262
             net/http.(*Server).Serve /home/wfender/.gvm/gos/go1.17.12/src/net/http/server.go:3002
             net/http.(*Server).ListenAndServe /home/wfender/.gvm/gos/go1.17.12/src/net/http/server.go:2931
             sigs.k8s.io/apiserver-network-proxy/cmd/agent/app.(*Agent).runHealthServer.func3.1 /home/wfender/Projects/network-proxy/src/apiserver-network-proxy/cmd/agent/app/server.go:110
             sigs.k8s.io/apiserver-network-proxy/cmd/agent/app.(*Agent).runHealthServer.func3 /home/wfender/Projects/network-proxy/src/apiserver-network-proxy/cmd/agent/app/server.go:115
             runtime/pprof.Do /home/wfender/.gvm/gos/go1.17.12/src/runtime/pprof/runtime.go:40
-----------+-------------------------------------------------------

Reply:
I had been trying to minimize the change which was why I didn't turn these into named function.
I do agree that it would be a nicer stack and easier to understand code. 
I'll go ahead and make that change.